lib/Sema/CSGen.cpp

```diff
@@ -3947,7 +3947,9 @@ namespace {
 
     Type visitTapExpr(TapExpr *expr) {
       DeclContext *varDC = expr->getVar()->getDeclContext();
-      assert(varDC == CS.DC || (varDC && isa<AbstractClosureExpr>(varDC)) &&
+      assert((varDC == CS.DC ||
+              (varDC && isa<AbstractClosureExpr>(varDC)) ||
+              varDC->isChildContextOf(CS.DC)) &&
```

Comment:
can varDC ever be null? If it can then you'll crash here, otherwise if not, you don't need the null check before the isa

Reply:
AH, thank you!